### PR TITLE
Add BERT resources

### PR DESCRIPTION
**Pull request by stevhliu**
This PR kicks off #19848 and adds official resources for BERT to the BERT model doc page. The resources are grouped by the types of tasks you can use the model for as well as using it for some other applications like inference and deployment.

If possible, I think it'd also be cool to use the task icons we use on the Hub and Tasks page (see below for example). What do you think?

Screen Shot 2022-10-24 at 1 26 25 PM

### CODE REVIEW

File: docs/source/en/model_doc/bert.mdx [hunk 1]
```diff
@@ -41,6 +41,54 @@ Tips:
...
+- A notebook for [Finetuning BERT (and friends) for multi-label text classification](https://colab.research.google.com/github/NielsRogge/Transformers-Tutorials/blob/master/BERT/Fine_tuning_BERT_(and_friends)_for_multi_label_text_classification.ipynb).
+- [`BertForSequenceClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/text-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/text_classification.ipynb).
+- [`TFBertForSequenceClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/text-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/text_classification-tf.ipynb).
+- [`FlaxBertForSequenceClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/flax/text-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/text_classification_flax.ipynb).
+
+**NER**
```
Comment (sgugger) [inline docs/source/en/model_doc/bert.mdx:53]: Let's use Token classification here, as NER is just one token classification task.

File: docs/source/en/model_doc/bert.mdx [hunk 2]
```diff
@@ -41,6 +41,54 @@ Tips:
...
+- [`BertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/token-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification.ipynb).
+- [`TFBertforTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/tensorflow/token-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/token_classification-tf.ipynb).
+- [`FlaxBertForTokenClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/flax/token-classification).
+- [Token classification](https://huggingface.co/course/chapter7/2?fw=pt) chapter of the 🤗 Hugging Face Course.
+
+**Language modeling**
```
Comment (sgugger) [inline docs/source/en/model_doc/bert.mdx:61]: Let's be clear here about which language modeling it is (masked, not causal)

**Review by sgugger — APPROVED** [review]
Great, thanks a lot for kickstarting this effort! Re-using the task buttons from the hub sounds like a great idea!

**Comment by LysandreJik** [comment]
I really like this!

File: docs/source/en/model_doc/bert.mdx [hunk 3]
```diff
@@ -41,6 +41,54 @@ Tips:
 
 This model was contributed by [thomwolf](https://huggingface.co/thomwolf). The original code can be found [here](https://github.com/google-research/bert).
 
+## Resources
+
+**Text classification**
+- A blog post on [BERT Text Classification in a different language](https://www.philschmid.de/bert-text-classification-in-a-different-language).
+- A notebook for [Finetuning BERT (and friends) for multi-label text classification](https://colab.research.google.com/github/NielsRogge/Transformers-Tutorials/blob/master/BERT/Fine_tuning_BERT_(and_friends)_for_multi_label_text_classification.ipynb).
```
Comment (NielsRogge) [inline docs/source/en/model_doc/bert.mdx:48]: I would perhaps add non-official example scripts/notebooks at the bottom?

Comment (stevhliu) [reply docs/source/en/model_doc/bert.mdx:48]: Sure thing, I'll pull some from our community resources and add an emoji or something next to it to indicate they are non-Hugging Face scripts/notebooks! 

I'll also add at the bottom a note saying you can open a PR to submit a notebook/resource (as long as its significant and isn't a duplicate of something else) to be added to the model docs. What do you think?

**Comment by stevhliu** [comment]
We don't have an icon for multiple choice, so would it be ok to add it under question answering (since its kind of a variant of question answering)?

**Comment by sgugger** [comment]
It's more a variant of sequence classification technically.

**Comment by stevhliu** [comment]
Ah ok! Would it be too confusing to include it under sequence classification then? We can also just leave it as is 😄

**Pull request merged**